packages/cli/src/commands/org/config/preview.spec.ts

```diff
@@ -0,0 +1,218 @@
+jest.mock('../../../lib/config/config');
+jest.mock('../../../hooks/analytics/analytics');
+jest.mock('../../../hooks/prerun/prerun');
+jest.mock('../../../lib/platform/authenticatedClient');
+jest.mock('../../../lib/snapshot/snapshot');
+jest.mock('../../../lib/snapshot/snapshotFactory');
+jest.mock('../../../lib/project/project');
+jest.mock('@oclif/errors');
+
+import {mocked} from 'ts-jest/utils';
+import {test} from '@oclif/test';
+import {Project} from '../../../lib/project/project';
+import {join} from 'path';
+import {cwd} from 'process';
+import {Config} from '../../../lib/config/config';
+import {SnapshotFactory} from '../../../lib/snapshot/snapshotFactory';
+import {Snapshot} from '../../../lib/snapshot/snapshot';
+import {warn, error} from '@oclif/errors';
+
+const mockedSnapshotFactory = mocked(SnapshotFactory, true);
+const mockedConfig = mocked(Config);
+const mockedProject = mocked(Project);
+const mockedWarn = mocked(warn);
+const mockedError = mocked(error);
+const mockedConfigGet = jest.fn();
+const mockedDeleteTemporaryZipFile = jest.fn();
+const mockedDeleteSnapshot = jest.fn();
+const mockedSaveDetailedReport = jest.fn();
+const mockedRequiresSynchronization = jest.fn();
+const mockedPreviewSnapshot = jest.fn();
+const mockedLastReport = jest.fn();
+
+const mockProject = () => {
+  mockedProject.mockImplementation(
+    () =>
+      ({
+        compressResources: () =>
+          Promise.resolve(join('path', 'to', 'resources.zip')),
+        deleteTemporaryZipFile: mockedDeleteTemporaryZipFile,
+      } as unknown as Project)
+  );
+};
+
+const mockConfig = () => {
+  mockedConfigGet.mockReturnValue(
+    Promise.resolve({
+      region: 'us-east-1',
+      organization: 'foo',
+      environment: 'prod',
+    })
+  );
+
+  // TODO: use prototype
+  mockedConfig.mockImplementation(
+    () =>
+      ({
+        get: mockedConfigGet,
+      } as unknown as Config)
+  );
+};
+
+const mockSnapshotFactory = async (validResponse: any) => {
```

Comment:
If the type is not really important here, I would put `unknown` (it's a bit better than any)